crates/ruff_linter/src/rules/flake8_pie/rules/no_unnecessary_pass.rs

```diff
@@ -51,28 +50,23 @@ impl AlwaysFixableViolation for UnnecessaryPass {
 
 /// PIE790
 pub(crate) fn no_unnecessary_pass(checker: &mut Checker, body: &[Stmt]) {
-    let [first, second, ..] = body else {
-        return;
-    };
-    // This only catches the case in which a docstring makes a `pass` statement
-    // redundant. Consider removing all `pass` statements instead.
-    if !is_docstring_stmt(first) {
+    if body.len() < 2 {
         return;
     }
 
-    // The second statement must be a `pass` statement.
-    if !second.is_pass_stmt() {
-        return;
-    }
-
-    let mut diagnostic = Diagnostic::new(UnnecessaryPass, second.range());
-    if checker.patch(diagnostic.kind.rule()) {
-        let edit = if let Some(index) = trailing_comment_start_offset(second, checker.locator()) {
-            Edit::range_deletion(second.range().add_end(index))
-        } else {
-            fix::edits::delete_stmt(second, None, checker.locator(), checker.indexer())
-        };
-        diagnostic.set_fix(Fix::automatic(edit));
-    }
-    checker.diagnostics.push(diagnostic);
+    body.iter()
+        .filter(|stmt| stmt.is_pass_stmt())
+        .for_each(|stmt| {
+            let mut diagnostic = Diagnostic::new(UnnecessaryPass, stmt.range());
+            if checker.patch(diagnostic.kind.rule()) {
+                let edit =
+                    if let Some(index) = trailing_comment_start_offset(stmt, checker.locator()) {
+                        Edit::range_deletion(stmt.range().add_end(index))
+                    } else {
+                        fix::edits::delete_stmt(stmt, None, checker.locator(), checker.indexer())
+                    };
+                diagnostic.set_fix(Fix::automatic(edit));
+            }
+            checker.diagnostics.push(diagnostic);
+        });
```

Comment:
I'm genuinely unsure what's faster: this, or passing the `pass` statement itself to `no_unnecessary_pass`, and then using `traversal::suite` to find the containing body and see if it's of length one or not.

Reply:
Like this is probably faster if `pass` statements are common, but they're relatively rare.